### PR TITLE
Feature/orca 261 Research spike card for automating integration tests

### PR DESCRIPTION
**Pull request by rizbih**
## Summary of Changes

Please write a sentence or two summarizing the proposed change.

Addresses [ORCA-261: Research spike card for automating integration tests](https://bugs.earthdata.nasa.gov/browse/ORCA-261)

## Changes

* created website/docs/developer/research/research-bamboo-integration-tests.md
* added https://github.com/nasa/cumulus-orca/blob/feature/ORCA-test-bamboo/bin/deployment-rds.sh
* added https://github.com/nasa/cumulus-orca/blob/feature/ORCA-test-bamboo/bin/deployment-cumulus-orca.sh
* updated https://github.com/nasa/cumulus-orca/blob/feature/ORCA-test-bamboo/bamboo-specs/bamboo.yaml

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

tested deploying some modules in aws via bamboo.

## Checklist:
- [ ] I have made corresponding changes to the necessary documentation 
    - [ x] Development documentation

### CODE REVIEW

**Review by AndrewDorn — APPROVED** [review]

File: website/docs/developer/research/research-bamboo-integration-tests.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+title: Research Notes on running integration tests in bamboo CI/CD
+description: Research Notes on deploying Cumulus and ORCA in bamboo CI/CD and running integration tests.
+---
+
+## Deploying terraform modules via bamboo
+AWS resources can be deployed via Bamboo pipeline by first adding the following to a script and running that in a Bamboo task.
+
+```bash
+#configure aws 
+aws configure set aws_access_key_id $bamboo_AWS_ACCESS_KEY_ID
```
Comment (bhazuka) [inline website/docs/developer/research/research-bamboo-integration-tests.md:12]: If you have the values for AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, and AWS_DEFAULT_REGION you should not need to run the aws configure commands just do an export.

```
export AWS_ACCESS_KEY_ID=$bamboo_AWS_ACCESS_KEY_ID
export AWS_SECRET_ACCESS_KEY=$bamboo_AWS_SECRET_ACCESS_KEY
export AWS_DEFAULT_REGION=$bamboo_AWS_DEFAULT_REGION
```

See https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html

File: website/docs/developer/research/research-bamboo-integration-tests.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+- https://bugs.earthdata.nasa.gov/browse/ORCA-389
+- https://bugs.earthdata.nasa.gov/browse/ORCA-390
+- https://bugs.earthdata.nasa.gov/browse/ORCA-391
+- https://bugs.earthdata.nasa.gov/browse/ORCA-392
+- https://bugs.earthdata.nasa.gov/browse/ORCA-393
+
+
+##### References
+- https://github.com/nasa/cumulus/tree/master/bamboo
+- Slack communication in cumulus-internal channel
```
Comment (bhazuka) [inline website/docs/developer/research/research-bamboo-integration-tests.md:129]: Provide links to these chats or else remove it as it does not provide us with anything useful otherwise.

I would also provide information/links on how Cumulus and others do it as a reference source like GHRC and LP from your research. I would include links to their scripts/repos.

File: website/docs/developer/research/research-bamboo-integration-tests.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+      interpreter: SHELL
+      scripts:
+      - |-
+        #!/bin/bash
+        chmod +x bin/deployment-cumulus-orca.sh
+        bin/deployment-cumulus-orca.sh
+```
+
+## Future directions
+
```
Comment (bhazuka) [inline website/docs/developer/research/research-bamboo-integration-tests.md:116]: I did not see any information on running in a Docker container, or the script snippets from Abdelhack on ways to better manage terraform variables needed for deployment using env files. I think I would add some basic information about that to help us understand as we move forward so things can be better/easier maintained.

**Review by bhazuka — CHANGES_REQUESTED** [review]
Couple of things to look at and add information on.

File: website/docs/developer/research/research-bamboo-integration-tests.md (hunk 2 above)
Comment (rizbih) [reply website/docs/developer/research/research-bamboo-integration-tests.md:129]: added more references

File: website/docs/developer/research/research-bamboo-integration-tests.md (hunk 1 above)
Comment (rizbih) [reply website/docs/developer/research/research-bamboo-integration-tests.md:12]: yes tested and looks good. 
updated bin/deployment-cumulus-orca.sh and bin/deployment-rds.sh

File: website/docs/developer/research/research-bamboo-integration-tests.md (hunk 3 above)
Comment (rizbih) [reply website/docs/developer/research/research-bamboo-integration-tests.md:116]: added links to GHRC dockerfile repo.I did create a card to look more into this. See [ORCA-393](https://bugs.earthdata.nasa.gov/browse/ORCA-393)

File: website/docs/developer/research/research-bamboo-integration-tests.md (hunk 1 above)
Comment (bhazuka) [reply website/docs/developer/research/research-bamboo-integration-tests.md:12]: You may also want to update your documentation as well in this document to show them as exports.

File: website/docs/developer/research/research-bamboo-integration-tests.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+The first step will be to finish deploying all cumulus tf modules via Bamboo into sandbox account. Once that is completed, automation scripts for running the integrations tests need to be created. The original plan `ORCA Integrator` in bamboo will need to be updated with the same changes made in `prototype-demo` plan's `feature/ORCA-test-bamboo` branch so that the `develop` branch is updated. In addition, older resources should be deleted from AWS once tests are validated. 
+
+The details of the DockerFile as well as the scripts currently used by GHRC to deploy their repository to Bamboo CI/CD can be found [here](https://github.com/ghrcdaac/GHRC-TF-CICD) which will be useful to look into later on. Moreover, the environment variables in bamboo can be exported using env files to better manage terraform variables needed for deployment. See [ORCA-393](https://bugs.earthdata.nasa.gov/browse/ORCA-393) for more information on this.
+
+
+Some of the cards created to finish the task include:
+- https://bugs.earthdata.nasa.gov/browse/ORCA-388
```
Comment (bhazuka) [inline website/docs/developer/research/research-bamboo-integration-tests.md:123]: May want to put a brief summary of each card so we do not have to click each link to look them up.

File: website/docs/developer/research/research-bamboo-integration-tests.md (hunk 1 above)
Comment (rizbih) [reply website/docs/developer/research/research-bamboo-integration-tests.md]: oops forgot to do that. Updated

**Review by bhazuka — CHANGES_REQUESTED** [review]
Couple of minor changes/additions.

File: website/docs/developer/research/research-bamboo-integration-tests.md (hunk 4 above)
Comment (rizbih) [reply website/docs/developer/research/research-bamboo-integration-tests.md]: added summary to each card

File: website/docs/developer/research/research-orca-delete-functionality.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
+---
```
Comment (bhazuka) [inline website/docs/developer/research/research-orca-delete-functionality.md:1]: This shouldn't be a part of this PR, It should be added to a different PR.

**Review by bhazuka — CHANGES_REQUESTED** [review]
Need to remove delete functionality file from this PR

File: website/docs/developer/research/research-orca-delete-functionality.md (hunk 5 above)
Comment (rizbih) [reply website/docs/developer/research/research-orca-delete-functionality.md]: sorry got uploaded by mistake

**Review by bhazuka — APPROVED** [review]
